app/Presenters/Admin/DigitalPublicationArticlePresenter.php

```diff
@@ -57,6 +59,21 @@ public function getBrowseMoreLink($showAll = false)
         return '';
     }
 
+    public function countAllChildren($entity)
+    {
+        $count = 0;
+        foreach ($entity->children as $child) {
+            if ($child->type !== DigitalPublicationArticleType::Grouping) {
+                $count += 1;
+            }
+
+            if ($child->children && $child->children->count() > 0) {
+                $count += $this->countAllChildren($child);
+            }
+        }
+        return $count;
+    }
+
```

Comment:
The NestedSet package includes a `descendants` method that retrieves all children recursively: https://github.com/lazychaser/laravel-nestedset?tab=readme-ov-file#ancestors-and-descendants.

So rather than writing your own recursive method here, I think you can call `$entity->descendants`, `->filter()` out items that have the type Grouping, then count the results.